compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -956,14 +956,42 @@ impl<'a> Context<'a> {
         if self.handle_constant_index(instruction, dfg, index, array, store_value)? {
             return Ok(());
         }
-
-        let (new_index, new_value) =
-            self.convert_array_operation_inputs(array, dfg, index, store_value)?;
+        let mut offset = None;
+        let array_id = dfg.resolve(array);
+        let array_typ = dfg.type_of_value(array_id);
+        if dfg.instruction_results(instruction).len() == 1 {
+            let result_type = dfg.type_of_value(dfg.instruction_results(instruction)[0]);
+            if let Type::Array(item_type, _) = array_typ {
```

Comment:
```suggestion
            if let Type::Array(item_type, _) | Type::Slice(types) = array_typ {
```
We probably want to look at slices too. We banned nested slices but a slice can have nested arrays

Reply:
Done